domino/domino.py

```diff
@@ -122,14 +122,30 @@ def runs_start_blocking(self, command, isDirect=False, commitId=None,
                         Maximum number of seconds to wait for a task to
                         complete. If this threshold is exceeded, an exception
                         is raised.
+
+        retry_count : int (Optional)
+                        Maximum number of retry to do while polling
+                        (in-case of transient httpd errors). If this
+                        threshold exceeds, an exception is raised.
         """
         run_response = self.runs_start(command, isDirect, commitId, title,
                                        tier, publishApiEndpoint)
         run_id = run_response['runId']
 
         poll_start = time.time()
+        current_retry_count = 0
         while True:
-            run_info = self.get_run_info(run_id)
+            try:
+                run_info = self.get_run_info(run_id)
```

Comment:
Do we want to reset the `current_retry_count` to 0 here? I think this would make more sense to the end user. I think the behavior we want is a successful retry resets the count rather than we have the right to up to `retry_count` non-consecutive failures.

Reply:
Yes, it makes more sense.